content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md

```diff
@@ -283,6 +272,21 @@ There are some implicit conventions worth noting here:
 
     {{< codenew file="pods/topology-spread-constraints/one-constraint-with-nodeaffinity.yaml" >}}
 
+Since the topology domains are determined from the nodes that satisfy `spec.nodeSelector` or `spec.affinity.nodeAffinity`, topology domains that have zero nodes are excluded from calculations. This could lead to a problem when a node pool (or node group) is scaled to zero nodes and the user is expecting them to scale up, because, those topology domains won't be considered until there is at least one node in them.
```

Comment:
The topology domains are determined from the nodes that exist in the cluster, not just the ones that match.

I would say something like: `The scheduler doesn't have prior knowledge of all the zones or other topology domains that a cluster has. They are determined from the existing nodes in the cluster`.

Then you describe the problem:

`This could lead to a problem in autoscaled clusters, when a node pool ....`

Reply:
> The topology domains are determined from the nodes that exist in the cluster, not just the ones that match.

Nodes matching nodeSelector/nodeAffinity is the exact searching scope, isn't it? As nil nodeSelector/nodeAffinity means "match any node", say nodes that exist in the cluster.

Reply:
Uhm, you are right. Only the matching nodes count.